CMakeLists.txt

```diff
@@ -17,17 +16,70 @@ endif()
 find_package(ament_cmake REQUIRED)
 find_package(rosidl_default_generators REQUIRED)
 
+set(protobuf_MODULE_COMPATIBLE TRUE)
+find_package(Protobuf CONFIG REQUIRED)
+message(STATUS "Using protobuf ${Protobuf_VERSION}")
+
+find_package(gRPC CONFIG REQUIRED)
+message(STATUS "Using gRPC ${gRPC_VERSION}")
+
+file(MAKE_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}/proto)
+add_custom_command(
+      OUTPUT "${CMAKE_CURRENT_BINARY_DIR}/proto/shisen.pb.cc"
+            "${CMAKE_CURRENT_BINARY_DIR}/proto/shisen.pb.h"
+            "${CMAKE_CURRENT_BINARY_DIR}/proto/shisen.grpc.pb.cc"
+            "${CMAKE_CURRENT_BINARY_DIR}/proto/shisen.grpc.pb.h"
+      COMMAND $<TARGET_FILE:protobuf::protoc>
+      ARGS --grpc_out "${CMAKE_CURRENT_BINARY_DIR}/proto"
+        --cpp_out "${CMAKE_CURRENT_BINARY_DIR}/proto"
+        -I ${CMAKE_CURRENT_SOURCE_DIR}/proto
+        --plugin=protoc-gen-grpc=$<TARGET_FILE:gRPC::grpc_cpp_plugin>
+        ${CMAKE_CURRENT_SOURCE_DIR}/proto/shisen.proto
+      DEPENDS ${CMAKE_CURRENT_SOURCE_DIR}/proto/shisen.proto)
```

Comment:
Can you format this better?
- Use 2 spaces like the other lines.
- Put end bracket on new line like the other lines.
- Either put argument lists aligned with the argument key or follow 2 spaces tabulation.